src/DynamoCoreWpf/ViewModels/Core/PortViewModel.cs

```diff
@@ -416,6 +420,28 @@ private void AutoComplete(object parameter)
             wsViewModel.OnRequestNodeAutoCompleteSearch(ShowHideFlags.Show);
         }
 
+        // Handler to invoke Node autocomplete cluster
+        private void AutoCompleteCluster(object parameter)
+        {
+            var wsViewModel = node.WorkspaceViewModel;
+            wsViewModel.NodeAutoCompleteSearchViewModel.PortViewModel = this;
+
+            // If the input port is disconnected by the 'Connect' command while triggering Node AutoComplete, undo the port disconnection.
+            if (this.inputPortDisconnectedByConnectCommand)
+            {
+                wsViewModel.DynamoViewModel.Model.CurrentWorkspace.Undo();
+            }
+
+            // Bail out from connect state
+            wsViewModel.CancelActiveState();
+
+            if (wsViewModel.DynamoViewModel.IsNodeAutocompleteClusterEnabled)
+            {
+               MLNodeClusterAutoCompletionResponse results = wsViewModel.NodeAutoCompleteSearchViewModel.GetMLNodeClusterAutocompleteResults();
```

Comment:
Ok, so the results are still stored in `NodeAutoCompleteSearchViewModel` side and access by view extension. The view extension itself does not handle any communication layer to the service. Would you cover that in the design docs you are working on right now?

Reply:
So the results are actually stored in NodeAutocompleteCluster which is in NodeAutoCompleteView.xaml.cs. The API to call the service is in the search view model, which i plan to move to the NodeAutoCompleteViewModel.cs(introduced in this PR)